### DISCUSSION TITLE
How can I attach an EIP to an ECS using the terraform provider?

### DISCUSSION THREAD

**Discussion by TobiWo** [opentelekomcloud/terraform-provider-opentelekomcloud#2442] [Q&A] [answered]
Hi, I can't figure out how I can attach an EIP to an ECS.
As I only found one EIP resource I guess [this is what I need](https://registry.terraform.io/providers/opentelekomcloud/opentelekomcloud/latest/docs/resources/vpc_eip_v1). But how can I connect this to an [ecs v1](https://registry.terraform.io/providers/opentelekomcloud/opentelekomcloud/latest/docs/resources/ecs_instance_v1)?

Checking the ui an ecs has an nic parameter where private and public IPs are attached to. Ok, but when I check the [network resource](https://registry.terraform.io/providers/opentelekomcloud/opentelekomcloud/latest/docs/resources/networking_network_v2) I see that there is no way to add an eip. Maybe somehow to the [value_specs](https://registry.terraform.io/providers/opentelekomcloud/opentelekomcloud/latest/docs/resources/networking_network_v2#value_specs)? I have no clue. Any help is highly appreciated.

**Comment by artem-lifshits** [comment, marked as answer]
Here's example config:
```
resource "opentelekomcloud_networking_floatingip_v2" "this" {
  pool = "admin_external_net"
}

resource "opentelekomcloud_ecs_instance_v1" "instance_1" {
  name     = "server_1"
  image_id = var.image_id
  flavor   = "s2.medium.1"
  vpc_id   = var.vpc_id

  nics {
    network_id = var.network_id
  }

  data_disks {
    size = 10
    type = "SSD"
  }

  password                    = "Password@123"
  security_groups             = [var.security_group_id]
  availability_zone           = "eu-de-01"
  auto_recovery               = false
  delete_disks_on_termination = true
}

resource "opentelekomcloud_networking_floatingip_associate_v2" "this" {
  floating_ip = opentelekomcloud_networking_floatingip_v2.this.address
  port_id     = opentelekomcloud_ecs_instance_v1.instance_1.nics.0.port_id
}
```
